contracts/protocol/tokenization/ampl/AAmplToken.sol

```diff
@@ -9,55 +9,126 @@ import {IAToken} from '../../../interfaces/IAToken.sol';
-     Total aAMPL suppxly increases by exactly x.
-
+     Total aAMPL supply increases by exactly x.
   2) At any time, user can burn x aAMPLs for x AMPLs.
      Total aAMPL supply decreases by exactly x.
-
   3) At any time, userA can transfer x aAMPLs to userB.
      userA's aAMPL balance reduces by X.
      userB's aAMPL balance increases by X.
      Total aAMPL supply exactly remains same.
-
   4) When AMPL's supply rebases, only the 'unborrowed' aAMPL should rebase.
-     => Say there are 1000 aAMPL, and 200 AMPL is lent out. AMPL expands by 10%.
+      * Say there are 1000 aAMPL, and 200 AMPL is lent out. AMPL expands by 10%.
         Thew new aAMPL supply should be 1080 aAMPL.
-     => Say there are 1000 aAMPL, and 200 AMPL is lent out. AMPL contracts by 10%.
+      * Say there are 1000 aAMPL, and 200 AMPL is lent out. AMPL contracts by 10%.
         Thew new aAMPL supply should be 920 aAMPL.
-
   5) When AMPL's supply rebases, only the part of the balance of a user proportional to  the available liquidity ('unborrowed') should rebase.
-    => Say a user has deposited 1000 AMPL and receives 1000 aAMPL, and
-       20% of the total underlying AMPL is lent out. AMPL expands by 10%.
-       The new aAMPL user balance should be 1080 aAMPL.
-    => Say a user has deposited 1000 AMPL and receives 1000 aAMPL, and
-       20% of the total underlying AMPL is lent out. AMPL contracts by 10%.
-       The new aAMPL supply should be 920 aAMPL.
-
-  On mint and burn a private variable `_totalScaledAMPLSupply` keeps track of
-    the scaled AMPL principal deposited.
-*/
-contract AAmplToken is VersionedInitializable, IncentivizedAAmplERC20, IAToken {
+      * Say a user has deposited 1000 AMPL and receives 1000 aAMPL, and
+        20% of the total underlying AMPL is lent out. AMPL expands by 10%.
+        The new aAMPL user balance should be 1080 aAMPL.
+      * Say a user has deposited 1000 AMPL and receives 1000 aAMPL, and
+        20% of the total underlying AMPL is lent out. AMPL contracts by 10%.
+        The new aAMPL supply should be 920 aAMPL.
+  6) The totalSupply of aAMPL should always be equal to (total AMPL held in the system) + (total principal borrowed denominated in AMPL) + (interest)
+
+
+  Generic aToken:
+
+  ATokens have a private balance and public balance.
+
+  ```
+    # _balances[u] and _totalSupply from contract storage.
+    Balance(u) = _balances[u] . I
+    TotalSupply(supply) = _totalSupply . I
+  ```
+
+  The internal (fixed-supply) balance and totalSupply are referred to as balanceScaled and totalSupplyScaled
+  and correspond to the principal deposited.
+
+  The external (elastic-supply) balance and totalSupply correspond to the principal + interest.
+
+
+  AAMPL:
+
+  AAMPL tokens have an additional scaling factor (multiplier) on top of the existing AToken structure.
+  Thus they have 2 private balances and 1 public balance.
+
+    * The internal (fixed-supply) balance and totalSupply are referred to as balanceScaled2 and
```

Comment:
Can you replace 1 and 2 with words?
Elastic,fixed,internal,external,gons,AMPLs,aAMPLs